### ISSUE TITLE
showing XXX when using a custom league (european soccer for example)

### ISSUE THREAD

**Issue by tretabyte** [vasqued2/ha-teamtracker-card#19]
Hi,

I would like to have a way to properly have the Leagues name on the top of the card, either using a regular manual input Title or picking the value from the API but right now if we use ```show_league: true``` while on custom league like european soccer, the top card will only show XXX

**Comment by vasqued2** [comment]
Good catch.  It will require changes to both the sensor and card to fix.  I'll fix it in v0.6.0.

**Comment by RickxZen** [comment]
I just changed XXX in custom config to Eredivisie and it shows like this:
![afbeelding](https://user-images.githubusercontent.com/48241736/201607115-dffc3e6d-578e-47f6-b5e8-bd02baa961b9.png)

**Comment by vasqued2** [comment]
There are a couple different ways to handle this.  Not sure what you all are thinking.

1.  Make no change - Inputting an unsupported league id behaves the same way (although unintentionally) as inputting XXX and allows you to input a `sport_path` and `league_path`.  And then the card uses whatever value you specified as the league id like RickxZen discovered.
2. Pull the official league names back from the API and use either the name, abbreviation, or midsizeName.  Here are two examples of the values returned.
```
"name":"Dutch Eredivisie",
"abbreviation":"Erediv",
"midsizeName":"NED.1"
```
```
"name":"FIFA World Cup",
"abbreviation":"WC",
"midsizeName":"FIFA.WORLD"
```
My initial inclination is to document that you can use values other than XXX as the league ID as long as you specify the `sport_path` and `league_path`, but interested in your thoughts and if you would prefer one of the other values.

**Comment by tretabyte** [comment]
Hi,
why not a mix of both, pull the info from the API by default using the name/abbreviation and give the possibility of using a customizable name (just option given in the card), something like custom_name

**Comment by vasqued2** [comment]
I hadn't thought of the option in the card.  I think that's what I'll do.  Easy and gives most control to the user.  Thanks.

**Comment by tretabyte** [comment]
Hi
Just noticed the card_title option but it doesn't seem to be working with custom leagues or am I doing something wrong?

**Comment by tretabyte** [comment]
Hi,
got the update and it works perfectly!
thank you very much.

**Comment by tretabyte** [comment]
question, when we have no games within the time window the information gets pulled, it shows that smaller card with "no upcoming games", is there a way to change the XXX to whatever we use, maybe enable the name propertie or pulling the valid name from the API?
the card with games in it is perfect, the one without games is what bothers me with those XXX (I tried create a new one and adding some name instead of XXX to trigger the custom confirmation but kept getting invalid config and wouldn't move forwared)
![image](https://user-images.githubusercontent.com/60727013/203312667-d2595e24-a3f0-4475-8188-201c8e47b125.png)

**Comment by vasqued2** [comment]
The XXX is because it helps me troubleshoot issues quickly.  Making your change makes sense and will clean up the card, but I want to spend some time figuring out how I can still make it easy to troubleshoot common problems.

By far, the biggest problem people ask about is sensor unavailable or games not found.  Most of the time it's because they set up the sensor wrong, but it's always possible there is a coding problem that I need to check out.  I use the info on this card to quickly triage out the user errors.

People who set up the sensor wrong generally take a screen shot but their ability to turn on debugging, check logs, and take other steps are limited.  Some don't even know how to look at the state and attribute values in Dev Tools.

To date, my focus on the Not Found card has been purely to help w/ debugging, not on making the card look pretty.  Let me think about how I can do both.

**Comment by vasqued2** [comment]
I'm open to suggestions

**Comment by vasqued2** [comment]
Were you picturing something like this, where "Liga Portugal" is the card_title?
![BEN](https://user-images.githubusercontent.com/17114183/204162063-aba35a99-b2db-4f49-95f6-80992ccf7fe9.jpg)

**Comment by tretabyte** [comment]
Hi,

yeah that would be perfect.

One thing I think needs to be reviewed is the way you pick the information to populate the configuration, you say to "See Custom API Configuration section below on how to set up additional sports/leagues if you know the ESPN API." which for me would be the tricky part of all this.
Anything that isn't "default" is always a pain to figure out.
Oh another thing I think it's very important to highlight, apparently ESPN doesn't always shows all matches, per team or even per league. A good example is the one you pasted above, although it says there isn't any upcoming game through 12/01, yesterday there was one actually that didn't show in the website therefore the integration didn't pull the information, this is obviously an issue with ESPN not with your integration but it would be important to mention.

**Comment by vasqued2** [comment]
Thanks.  I'll include the changes in the next release.  

Feel free to update or add a page to the Wiki on how to help others w/ the Custom API configurations and I'll link to it in the README.  I don't know an easy way to explain it because there is no authoritative source for the APIs and it seems like someone is always finding more.  It's a bunch of trial and error - or in your words - "a pain to figure out".  Anything you can add to the Wiki to remove some of that pain would be a welcome addition!

The reason the game yesterday didn't show up is because it wasn't Liga Portugal game (looks like the league is on break through the World Cup, as are most leagues).  It was part of something called the Allianz Cup.  I assume you know more about what this is than I would.  Maybe there is an API for that but you would have to look and find it, which gets back to your first point. 

In the end, I created the Custom API Configuration to let people who could figure out the API themselves to do so.  It's actually gotten easier as more people have found more resources which are included in the Wiki, but again, it's not always straight-forward, and I don't think it's ever going to be.  Anything you can add to help is appreciated.

Thanks

**Comment by vasqued2** [comment]
The revamped layout from above is included in v0.5.4.  Let me know if you have issues after upgrading.
